app/src/features/Columns/ColumnSelect.tsx

```diff
@@ -41,11 +54,29 @@ const ColumnSelects = ({
   const { data: credentialOwners } = useApiOwnersListQuery({
     credential: credential?.[0].id,
   });
-  const { table, column } = pendingColumn;
-  const selectedOwner = credentialOwners?.[0];
+  const { table, column, owner: ownerId } = pendingColumn;
+  const selectedOwner = credentialOwners?.find(({ id }) => id === ownerId);
   const schema = selectedOwner?.schema as Record<string, string[]>;
+  const ownerTable =
+    table && selectedOwner
+      ? {
+          id: `${selectedOwner.id}/${table}`,
+          label: `${selectedOwner.name}/${table}`,
+        }
+      : undefined;
 
-  const tables = (schema && Object.keys(schema)) || [];
+  const tables = !credentialOwners
+    ? []
+    : credentialOwners.reduce((acc: { id: string; label: string }[], owner) => {
```

Comment:
Maybe that deserves its own function that you could call it here? You're doing several things here (extract the tables from the owner schema and transform the data to pass it to AutoComplete. It could me more explicit.